x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -290,9 +287,6 @@ void setUseAuto(boolean useAuto) {
     @Override
     public void onMaster() {
         isMaster = true;
-        if (running) {
-            mlMemoryTracker.asyncRefresh();
-        }
```

Comment:
@droberts195 This is the only thing I am not 100% sure of. 

The typical scenario for the master node to change is some sort of cluster mutation event. Meaning, nodes leaving/joining/etc. This increases the chance of some intermittent query failures due to shard issues, etc. 

We do scale up or down right away (regardless of memory freshness) in two scenarios:

- Scale UP with waiting jobs but no nodes. 
- Scale down (if delay satisfied) if no jobs but we do have nodes.

What do you think?

The end result of this change would be possibly slower autoscaling responsiveness in scenarios where:
 -  a job was NOT recently opened (opening a job kicks off an async refresh when memory is stale) 
     - This may also be caused by a race condition where the memory was "fresh" when the job checked, but stale by the time autoscaling was called. But, we are fairly liberal with our staleness check in autoscaling.
 - There is no room for said job
 - And there are existing ML nodes

Reply:
`MlMemoryTracker.onMaster()` also had a call to the same method, so previously we were calling it twice when a node became master.  I think it's worth leaving one of the calls in, but I'd prefer to leave the one in `MlMemoryTracker` in than this one.